### ISSUE TITLE
display-sessions shows nothing for clojurescript session

### ISSUE THREAD

**Issue by FongHou** [Olical/conjure#91]
Just a quick note. This only happens in develop branch. I guess due to commit 9bcf630e4a0664b86b3f26e55eee7f93a14d48a6.

Thanks!

**Comment by Olical** [comment]
Ah yes, I tweaked this while refactoring today, it'll just be a key in a
map I've typoed or something. I'll get that fixed, thanks for letting me
know!

**Comment by Olical** [comment]
Hmm, works on my machine :sweat_smile: could be when you're in a namespace that isn't loaded :thinking:

**Comment by Olical** [comment]
I made the session detection slightly more robust, maybe that'll do it for you?

**Comment by Olical** [comment]
I'm moving the session type into the list on develop so the "session type" mapping will go away anyway.

```
; --------------------------------------------------------------------------------
; Sessions (2):
; >1 - 23cd29a5-a392-4c82-83f1-63d5cc04c277 (Clojure)
;  2 - 409a9ad2-4218-485f-b4d4-aefac3b97ac2 (ClojureScript)
```

**Comment by Olical** [comment]
(so no more `st` mapping, just use `sl` to see all sessions and their types + which is currently selected)

**Comment by Olical** [comment]
https://asciinema.org/a/330985 here's how it should look now!

**Comment by FongHou** [comment]
Very nice! 

I still don't see clojurescript session in the list.

Steps to reproduce below screenshot.

1. Start a shadow-cljs server (shadow/dev app) ran successfully, no compile error.
2. Open a **cljs** file in nvim. Conjure auto connects, and display-sessions show one **Clojure** session. (I assume no auto-require occurs for cljs file because the session type is not Clojurescript yet).
3.  Mannally run command "ConjureShadowSelect app", then display-sessions show one session but session type becomes empty ().
4. Run eval buf with no error, display-sessions still show session type empty ().
5. Run clone-fresh-session, display-sessions show a new **Clojure** session.
6. Run select-session-interactive (select 2), got "conjure.client.clojure.nrepl error: ...gged/conjure/lua/conjure/client/clojure/nrepl/server.lua:288: attempt to concatenate field 'pretty-type' (a nil value)"

![image](https://user-images.githubusercontent.com/13973254/82157806-ae2a4480-9851-11ea-9362-d65134404bb2.png)

**Comment by Olical** [comment]
Oh! Okay so the error is a bug from my refactoring, I'll get that fixed
tomorrow! The lack of "ClojureScript" is super weird though since we're
doing the same thing. Would you be able to turn on debugging and show me
what you see?

:ConjureConfig clojure.nrepl/debug? true

It'll show us the internal nREPL messages, it should explain what's going
on.

**Comment by FongHou** [comment]
Just did exactly that :-) See debug log in above comment.

**Comment by Olical** [comment]
My bad, was replying through email and it was out of date. Thanks a lot! I'll get this fixed tomorrow :smile:

**Comment by Olical** [comment]
Ah hah! Yep, it's the newline! Good spot, I'll trim the output and we should be good to go! I wonder why I'm not seeing that newline in my project :thinking: ClojureScript can be a fickle beast sometimes.

Also the auto require just tries to `(require ...)` the namespace of the buffer you either just opened or the current buffer when you connect / piggieback to something new. If the require fails the error is thrown away, it doesn't care too much. So the error is acceptable in my opinion, the user will deliberately never see it.

You're probably right and auto require needs more work and thought though, it's something I just sort of added to address a big UX hole where people kept opening Conjure, trying to eval things and getting _really_ weird errors because I had switched them to a namespace that didn't exist. That means `clojure.core` doesn't exist which means you can't require anything!

**Comment by Olical** [comment]
Okay, pushed a fix for this where I trim the newline off. I've also improved the prettify of the session type so you won't get that error again if something like this happens in the future. There's no chance pretty-type can become a `nil` now, it'll either be the unrecognised type with a question mark on the end or `???` if it's `nli` (which should never happen).

**Comment by Olical** [comment]
Thank you so much for the detailed issue, you've been a huge help! :tada:

**Comment by Olical** [comment]
Another thing to note, if you're seeing `(???)` as the session type it might be because your JavaScript environment (browser or node) isn't connected yet, I've seen this myself. There's nothing I can do about that, every eval will fail until the environment is connected so there's no way I can tell it's ClojureScript since the check will throw an error.

**Comment by FongHou** [comment]
I'm happy to help. I just tried latest develop branch. Everything works beautifully. 

Regarding auto-require, I agree to your approach to keep it simple. 

Thank you for spending time improving conjure so quickly. You rock 💯 

Cheers!

**Comment by Olical** [comment]
Fixed in v3.2.0! :D
